### ISSUE TITLE
Method params types suggestion

### ISSUE THREAD

**Issue by harveysanders** [VeliovGroup/flow-router#104]
Hi!
Just a suggestion: 
Most of the `FlowRouter` methods have optional params.
```typescript
interface Router {
    // ....
    url: (path: string, params?: Param, qs?: QueryParam) => string;
    path: (path: string, params?: Param, qs?: QueryParam) => string;
    // ....
 }   
```
With these types, one needs to pass `undefined` or an empty placeholder value.
```typescript
const path = FlowRouter.path('a-path', undefined, {q: "a query"});
```

![image](https://user-images.githubusercontent.com/9354822/231816026-e93cdf30-22c6-4604-a80b-2b472189c5f3.png)

In my opinion, `undefined` represents values that are not yet set in the runtime, whereas if you explicitly need to set a value to a non-value, `null` should be used. With that said, I think it may make sense to use union types with `null` instead of the question mark so devs call the methods like this:
```typescript
const path = FlowRouter.path('a-path', null, {q: "a query"});
```

If you agree, I could submit a PR at some point.

Thanks for your time and work on this project!

**Comment by dr-dimitru** [comment]
Hello @harveysanders,

Pardon me for answering almost a year later, I agree, and love to add support for `null` along with `void 0`/`undefined` feel free so send a PR

**Comment by harveysanders** [comment]
@dr-dimitru no worries! I just submitted #106 for your consideration

**Comment by dr-dimitru** [comment]
Published as [`v3.10.1`](https://github.com/veliovgroup/flow-router/releases/tag/3.10.1)
@harveysanders thank you for contribution 🙏
